WordPress/gutenberg
# Site Editor: Enable copy/cut shortcut

Closes #43709
Related to #43352

## What?
This PR activates copy/cut shortcut in the site editor.

## Why?

When I was looking into the shortcut issue in Windows with #43352, I noticed that copy/cut shortcut is not available only for the site editor:

| Description                                 | Shortcut | Post Editor | Site Editor | Widget Editor | Customizer Widget |
| ------------------------------------------- | -------- | :---------: | :---------: | :-----------: | :---------------: |
| Cut the selected block(s).                  | Ctrl (Cmd) + X   |      ✅      |      ❌  |       ✅       |         ✅         |
| Copy the selected block(s).                 | Ctrl (Cmd) + C   |      ✅      |      ❌ |       ✅       |         ✅         |
| Paste the block(s) stored in the clipboard. | Ctrl (Cmd) + V   |      ✅      |      ✅      |       ✅       |         ✅         |

I was unable to find if this shortcut was intentionally opted out, but I think shortcuts should be as uniform as possible in all editor instances.

## How?

Added the `useClipboardHandler` hook to the site editor ref. Please advise me if you have any concerns about adding this hook.


## Testing Instructions

Confirm that the copy and cut shortcuts work in all of the following locations. It is also a good idea to check the paste shortcut at the same time.

- Template Editor
- List View in the Template Editor
- Template Parts Editor
- List View in the Template Editor

## Screenshots or screencast 

https://user-images.githubusercontent.com/54422211/201680237-c7df3ce0-31a1-4dae-8225-d9fc155c1a5c.mp4
